File: help_reviews.client/src/models/Report.js
import { Profile } from "./Account.js"

export class Report {
  constructor (data) {
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.title = data.title
    this.body = data.body
    this.rating = data.rating
    this.restaurantId = data.restaurantId
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
  }
}

File: help_reviews/Models/Report.cs
using System.ComponentModel.DataAnnotations;

namespace help_reviews.Models;

public class Report : RepoItem<int>
{
  public string Title { get; set; }
  public string Body { get; set; }
  [Range(1, 5)]
  // NOTE                         VVV Default value
  public int Rating { get; set; } = 1;
  public int RestaurantId { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}
